website/i18n/zh/docusaurus-plugin-content-blog/2021/11/23/cve-2021-43557-research-report.md

```diff
@@ -86,27 +88,29 @@ spec:
           regex_uri: ["/protected-service/(.*)", "/$1"]
 ```
 
-Let’s dive deep into it:
+`ApisixRoute` 可以做到以下内容：
+
+- 为 `public-service` 和 `private-service` 创建路由
+- 开启 `proxy-rewrite` 来删除前缀
+- 为 `protected-service` 配置了 `uri-blocker` 插件。这个插件会阻止任何以 `/protected-service` 开头的请求
 
-- It creates routes for `public-service` and `private-service`
-- There is `proxy-rewrite` turned on to remove prefixes
-- There is `uri-blocker` plugin configured for `protected-service`. It can look like mistake but this plugin it about to block any requests starting with `/protected-service`
+## 漏洞复现
 
-## Exploitation
+在这里，我使用的是 **2.10.0** 版本的 Apache APISIX。
 
-I’m using APISIX in version **2.10.0**.
+在 minikube 中接触到 Apache APISIX 路由是相当不方便的：`kubectl exec -it -n ${Apache APISIX namespace} ${Apache APISIX Pod name} -- curl --path-as-is http://127.0.0.1:9080/public-service/public -H 'Host: app.test'`。
```

Comment:
```suggestion
选择使用 minikube 来访问 Apache APISIX 路由资源：

```sh
$ kubectl exec -it -n ${Apache APISIX namespace} ${Apache APISIX Pod name} -- curl --path-as-is http://127.0.0.1:9080/public-service/public -H 'Host: app.test'
```

```